compiler/dyno/test/common.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021-2022 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "chpl/types/all-types.h"
+#include "chpl/uast/Identifier.h"
+#include "chpl/uast/Module.h"
+#include "chpl/uast/Record.h"
+#include "chpl/uast/Variable.h"
+
+namespace impl {
+
+using namespace chpl;
+using namespace resolution;
+using namespace types;
+using namespace uast;
+
+// Get the top-level module resulting from parsing the given string.
+const Module* parseModule(Context* context, const char* src);
+const Module* parseModule(Context* context, std::string& str);
+
+// assumes the last statement is a variable declaration for x
+// with an initialization expression.
+// Returns the type of the initializer expression.
+QualifiedType
+parseTypeOfXInit(Context* context, const char* program, bool requireTypeKnown = true);
+
+QualifiedType
+parseQualifiedTypeOfX(Context* context, const char* program);
+
+const Type*
+parseTypeOfX(Context* context, const char* program);
```

Comment:
it is OK to leave it as-is but `parseTypeOfX` is not a great name for this because it parses everything and then computes type of x. It does not, for example, just parse the type of x.

Reply:
`resolveTypeOfX` might be more accurate. In my mind, resolving something implies parsing more things (like the required definitions).

Reply:
Yes that would be great if you are up for changing the name

Reply:
Just to confirm: it seems to me like the rest of the `parse*` names should be changed (`parseQualifiedTypeOfX`, `parseTypeOfXInit`. That is what you had in mind, right?

Reply:
Yes, thank you